src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -238,6 +239,189 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
             return null;
         }
 
+#if XMLSERIALIZERGENERATOR
+        internal static class ThisAssembly
+        {
+            internal const string Version = "1.0.0.0";
+            internal const string InformationalVersion = "1.0.0.0";
+        }
+
+        private static string GenerateAssemblyId(Type type)
+        {
+            Module[] modules = type.Assembly.GetModules();
+            var list = new ArrayList();
+            for (int i = 0; i < modules.Length; i++)
+            {
+                list.Add(modules[i].ModuleVersionId.ToString());
+            }
+
+            list.Sort();
+            var sb = new StringBuilder();
+
+            for (int i = 0; i < list.Count; i++)
+            {
+                sb.Append(list[i].ToString());
+                sb.Append(",");
+            }
+
+            return sb.ToString();
+        }
+
+        internal static FileStream GenerateSerializerFile(XmlMapping[] xmlMappings, Type[] types, string defaultNamespace, Assembly assembly, Hashtable assemblies, string codePath)
+        {
+            var compiler = new Compiler();
+            try
+            {
+                var scopeTable = new Hashtable();
+                foreach (XmlMapping mapping in xmlMappings)
+                    scopeTable[mapping.Scope] = mapping;
+
+                var scopes = new TypeScope[scopeTable.Keys.Count];
+                scopeTable.Keys.CopyTo(scopes, 0);
+                assemblies.Clear();
+                var importedTypes = new Hashtable();
+
+                foreach (TypeScope scope in scopes)
+                {
+                    foreach (Type t in scope.Types)
+                    {
+                        compiler.AddImport(t, importedTypes);
+                        Assembly a = t.Assembly;
+                        string name = a.FullName;
+                        if (assemblies[name] != null)
+                        {
+                            continue;
+                        }
+
+                        if (!a.GlobalAssemblyCache)
+                        {
+                            assemblies[name] = a;
+                        }
+                    }
+                }
+
+                for (int i = 0; i < types.Length; i++)
+                {
+                    compiler.AddImport(types[i], importedTypes);
+                }
+
+                compiler.AddImport(typeof(object).Assembly);
+                compiler.AddImport(typeof(XmlSerializer).Assembly);
```

Comment:
Should we used the full qualified name, `System.Xml.Serialization.XmlSerializer`, here? Or is this `Microsoft.XmlSerializer.Generator.XmlSerializer`?